Fix clamp type promotion in inductor decomposition

yushangdi opened the pull request:
Summary: as title, the clamp type promotion should take min/max arg into consideration as well.

Test Plan:
```
buck run fbcode//caffe2/test/inductor:test_aot_inductor -- -r test_clamp_decomposition_cpu
python test/inductor/test_torchinductor.py -k test_clamp -v
```

Differential Revision: D75490124




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @desertfire @benjaminglass1

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -5766,6 +5766,22 @@ def wrapped(**kwargs):
+            def forward(self, x):
+                return x.clamp(min=1)
+
+        x = torch.randn(2, 3).to(torch.int64)
```
Let's use torch.randint to generate an integer tensor.

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -5766,6 +5766,22 @@ def wrapped(**kwargs):
         # compare against eager
         self.assertEqual(optimized(**model_kwargs), model(**model_kwargs))
 
+    def test_clamp_decomposition(self):
```
Please also add a test to test_torchinductor.py

chenyang78 reviewed (approved):
Thanks for the prompt fix!

yushangdi committed:
Fix clamp type promotion in inductor decomposition (#154471)

Summary:

as title, the clamp type promotion should take min/max arg into consideration as well.

Test Plan:
```
buck run fbcode//caffe2/test/inductor:test_aot_inductor -- -r test_clamp_decomposition_cpu
```

Reviewed By: desertfire, muchulee8

Differential Revision: D75490124
